python/tank/bootstrap/manager.py

```diff
@@ -202,30 +208,42 @@ def _set_bundle_cache_fallback_paths(self, paths):
     )
 
 
-    def set_progress_callback(self, callback):
+    def set_progress_callback(self, progress_callback):
         """
-        Specify a method to call whenever progress should be reported back.
+        Sets a function to call whenever progress of the bootstrap should be reported back.
+
+        This function should have the following signature::
```

Comment:
Yes, all the client provided callbacks are called from the main thread. Qt signals and slots are used between the worker and the main thread code.